src/aerovaldb/jsondb/cache.py

```diff
@@ -53,39 +57,88 @@ def remove(self, item: Hashable):
             self._deque.remove(item)
 
 
+class BaseCache(ABC):
+    def __init__(self):
+        pass
+
+    @property
+    @abstractmethod
+    def hit_count(self) -> int:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def size(self) -> int:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def miss_count(self) -> int:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get(self, key: str, *, bypass_cache: bool = False) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    def put(self, obj: str, *, key: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def invalidate_all(self) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def invalidate_entry(self, file_path: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_valid(self, key: str) -> bool:
+        raise NotImplementedError
+
+
 class CacheEntry(TypedDict):
     json: str
 
     last_modified: float
 
 
-class JSONLRUCache:
+class LRUFileCache(BaseCache):
     """
-    Implements an in-memory LRU cache for file content in aerovaldb.
+    Implements an in-memory LRU cache for file content.
     """
 
-    def __init__(self, *, max_size: int, asyncio: bool = False):
+    def __init__(self, *, max_size: int):
         """
         :param max_size : The maximum size of the cache in terms of number of entries / files.
 
         Files will be ejected based on least recently used, when full.
         """
-        self._asyncio = asyncio
+        self._hit_count: int = 0
+        self._miss_count: int = 0
         self._max_size = max_size
         self.invalidate_all()
```

Comment:
please don't invalidate_all() during init where nothing is set, this is confusing. Initialize all parameters in __init__ explicitly, e.g. self._queue